### DISCUSSION TITLE
Infinite redirect on Firebase Hosting after successful authentication

### DISCUSSION THREAD

**Discussion by leandigital** [gladly-team/next-firebase-auth#593] [Q&A]
Hi all.

I've got the whole authentication flow working perfectly locally and on Vercel. However, when using Firebase hosting (with Firebase function), immediately after user is authenticated, I'm seeing infinite requests to index.json, that seems to consist of the redirect object. However, I'm not seeing the redirect happening.

Chain of events (with log):

0. Init: `Setting config with provided value: {debug: true, appPageURL: '/', authPageURL: '/login', onLoginRequestError: ƒ, onLogoutRequestError: ƒ, …}`
2. Load login screen. `AuthUser set to: {id: null, email: null, emailVerified: false, phoneNumber: null, displayName: null, …}`
3. Authenticate. `Starting auth API request via tokenChangedHandler.`
4. Success. `Completed auth API request via tokenChangedHandler.`
5. Redirect chain/loop... `Redirecting to app.`

Locally this works with both firebase serve and firebase emulators:start. This also works on Vercel.

Any ideas of what may be going on here?

**Comment by kmjennison** [comment]
Not sure if this is what's happening in your case, but a possible cause is that Firebase hosting [only allows using a single `__session` cookie](https://firebase.google.com/docs/hosting/manage-cache#using_cookies). Issue: https://github.com/gladly-team/next-firebase-auth/issues/190
